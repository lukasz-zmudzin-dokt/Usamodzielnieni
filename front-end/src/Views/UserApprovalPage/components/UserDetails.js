import { Accordion, Card } from "react-bootstrap";
import React from "react";
import UserToApprove from "./UserToApprove";

const UserDetails = ({ users, activeUser, setActiveUser }) => {

    return (
        <Accordion>
            {users.map((user) =>
                <Card className="border-left-0 border-right-0 border-bottom-0" key={user.id}>
                    <Accordion.Toggle as={Card.Header} eventKey={user.id} onClick={e => setActiveUser(user.id)} >
                        {user.username} ({user.type})
                    </Accordion.Toggle>
                    <Accordion.Collapse eventKey={user.id}>
                        <UserToApprove user={user} key={user.id} activeUser={activeUser} />
                    </Accordion.Collapse>
                </Card>
            )}
        </Accordion>
    );
};

export default UserDetails;