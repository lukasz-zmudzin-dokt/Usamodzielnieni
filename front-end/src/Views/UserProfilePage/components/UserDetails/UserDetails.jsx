import React from "react";
import { ListGroup } from "react-bootstrap";
import UserProperty from "Views/UserProfilePage/components/UserProperty";

const UserDetails = (props) => {
  const { user, names } = props;
  const ignore = ["firstName", "lastName", "role", "chat_role"];

  const userProperties = Object.keys(user).map((key) => {
    return !ignore.find((i) => i === key) ? (
      <UserProperty key={key} user={user} property={key} names={names} />
    ) : null;
  });

  return (
    <ListGroup className="list-group-flush">
      {userProperties}
      {user.chat_role !== undefined ? (
        <UserProperty user={user} property={"chat_role"} names={names} />
      ) : null}
    </ListGroup>
  );
};

export default UserDetails;
