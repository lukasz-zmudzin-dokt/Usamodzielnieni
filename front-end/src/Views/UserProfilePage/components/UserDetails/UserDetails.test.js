import React from "react";
import { render } from "@testing-library/react";
import UserDetails from "Views/UserProfilePage/components/UserDetails";

const user = {
  username: "user1",
  role: "common",
  firstName: "Jan",
  lastName: "Kowalski",
  email: "jan.kowalski@pw.edu.pl",
  phoneNumber: "+48123456789"
};

const names = {
  role: {
    admin: "Administrator",
    employer: "Pracodawca",
    common: "Podopieczny"
  },
  property: {
    username: "Nazwa użytkownika",
    role: "Rola",
    firstName: "Imię",
    lastName: "Nazwisko",
    email: "E-mail",
    phoneNumber: "Numer telefonu"
  }
};

describe("UserDetails ", () => {

  it("should render correctly", () => {
    const { container } = render(
        <UserDetails user={user} names={names} />
    );
    expect(container).toMatchSnapshot();
  });
});