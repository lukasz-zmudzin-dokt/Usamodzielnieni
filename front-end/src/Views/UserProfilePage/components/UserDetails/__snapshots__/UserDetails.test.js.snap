// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails  should render correctly 1`] = `
<div>
  <div
    class="list-group-flush list-group"
  >
    <div
      class="property list-group-item"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-auto col-12"
        >
          <div
            class="property__title"
          >
            Nazwa u≈ºytkownika
            :
          </div>
        </div>
        <div
          class="col"
        >
          user1
        </div>
      </div>
    </div>
    <div
      class="property list-group-item"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-auto col-12"
        >
          <div
            class="property__title"
          >
            E-mail
            :
          </div>
        </div>
        <div
          class="col"
        >
          jan.kowalski@pw.edu.pl
        </div>
      </div>
    </div>
    <div
      class="property list-group-item"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-auto col-12"
        >
          <div
            class="property__title"
          >
            Numer telefonu
            :
          </div>
        </div>
        <div
          class="col"
        >
          +48123456789
        </div>
      </div>
    </div>
  </div>
</div>
`;
